site/profile/manifests/ccs/file_transfer.pp

```diff
@@ -61,6 +61,15 @@
   String $pkgurl_user = $profile::ccs::common::pkgurl_user,
   String $pkgurl_pass = $profile::ccs::common::pkgurl_pass,
 ) {
+  $parent = "${dirname($directory)}"
+
+  file { $parent:
```

Comment:
It would be a good idea to use [`ensure_resource()`](https://forge.puppet.com/modules/puppetlabs/stdlib/reference#ensure_resource-1) here to prevent a possible conflict with other profiles.

Reply:
Thanks for the comments; I have pushed a new version.